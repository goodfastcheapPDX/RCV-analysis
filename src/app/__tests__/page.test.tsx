import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Home from "../page";

describe("Home Page", () => {
  it("should match snapshot", () => {
    const { container } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
