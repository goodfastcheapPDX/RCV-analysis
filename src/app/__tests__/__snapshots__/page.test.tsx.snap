// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page > should match snapshot 1`] = `
<div
  class="container mx-auto px-4 py-12 max-w-4xl"
>
  <div
    class="text-center space-y-6 mb-12"
  >
    <div
      class="flex items-center justify-center gap-3"
    >
      <h1
        class="text-4xl md:text-5xl font-bold tracking-tight"
      >
        Ranked Choice Analysis
      </h1>
      <span
        class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-secondary text-secondary-foreground [a&]:hover:bg-secondary/90 text-xs"
        data-slot="badge"
      >
        alpha
      </span>
    </div>
    <p
      class="text-xl -foreground max-w-2xl mx-auto"
    >
      Represent who? Understanding proportional representation in ranked-choice elections.
    </p>
  </div>
  <div
    class="grid md:grid-cols-2 gap-6 mb-12"
  >
    <div
      class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm hover:shadow-lg transition-shadow"
      data-slot="card"
    >
      <div
        class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6"
        data-slot="card-header"
      >
        <div
          class="leading-none font-semibold"
          data-slot="card-title"
        >
          Explore Elections
        </div>
      </div>
      <div
        class="px-6"
        data-slot="card-content"
      >
        <p
          class="-foreground mb-4"
        >
          Browse available election data and start your analysis journey.
        </p>
        <a
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-primary underline-offset-4 hover:underline h-9 px-4 py-2 has-[>svg]:px-3 w-full"
          data-slot="button"
          href="/e"
        >
          Explore Elections
        </a>
      </div>
    </div>
    <div
      class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm hover:shadow-lg transition-shadow"
      data-slot="card"
    >
      <div
        class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6"
        data-slot="card-header"
      >
        <div
          class="leading-none font-semibold"
          data-slot="card-title"
        >
          Jump to Demo
        </div>
      </div>
      <div
        class="px-6"
        data-slot="card-content"
      >
        <p
          class="-foreground mb-4"
        >
          See the platform in action with Portland's 2024 City Council District 2 election.
        </p>
        <a
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-primary underline-offset-4 hover:underline h-9 px-4 py-2 has-[>svg]:px-3 w-full"
          data-slot="button"
          href="/demos"
        >
          View Demo
        </a>
      </div>
    </div>
  </div>
  <div
    class="bg-border shrink-0 data-[orientation=horizontal]:h-px data-[orientation=horizontal]:w-full data-[orientation=vertical]:h-full data-[orientation=vertical]:w-px my-8"
    data-orientation="horizontal"
    data-slot="separator"
    role="none"
  />
  <div
    class="text-center space-y-4"
  >
    <h2
      class="text-2xl font-semibold"
    >
      How It Works
    </h2>
    <div
      class="relative w-full rounded-lg border px-4 py-3 text-sm grid has-[>svg]:grid-cols-[calc(var(--spacing)*4)_1fr] grid-cols-[0_1fr] has-[>svg]:gap-x-3 gap-y-0.5 items-start [&>svg]:size-4 [&>svg]:translate-y-0.5 [&>svg]:text-current bg-card text-card-foreground"
      data-slot="alert"
      role="alert"
    >
      <div
        class="-foreground col-start-2 grid justify-items-start gap-1 text-sm [&_p]:leading-relaxed text-left"
        data-slot="alert-description"
      >
        <strong>
          CSV → Static Artifacts → Interactive UI
        </strong>
        <br />
        Upload Cast Vote Record (CVR) data in CSV format. Our analysis engine processes the data to generate static artifacts including STV tabulation results, coalition analysis, and vote transfer patterns. The interactive dashboard then provides comprehensive visualizations and insights into ranked-choice voting dynamics.
      </div>
    </div>
  </div>
</div>
`;
